src/libraries/Microsoft.PowerFx.Core/Functions/DefaultResourceElement.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    public abstract class BaseResourceElement
+    {
+        public abstract bool IsUri { get; }
+
+        public virtual Uri Uri => new Uri($"appres://blobmanager/{Handle.Handle}");
+
+        public ResourceHandle Handle { get; init; }
+
+        public abstract Task<string> GetAsStringAsync();
```

Comment:
>GetAsStringAsync [](http://example.com/codeflow?start=37&length=16)

A resource should be based on bytes, not strings.

Reply:
We can have multiple forms: byte[], string, base64string, Stream...